### PR TITLE
[S293] check for params in env, connection reset exception

### PR DESCRIPTION
**Pull request by saanobhaai**

### CODE REVIEW

**Comment by saanobhaai** [comment]
[task refactoring for pipeline](https://trello.com/c/w0Hpgoh8/293-task-refactoring-for-pipeline)

File: task_base.py [hunk 1]
```diff
@@ -43,10 +43,9 @@ def _set_inputs(self):
 
     def __init__(self, *args, **kwargs):
         _taskdate = datetime.now(timezone.utc).date()
+        _taskdatestr = kwargs.get("taskdate") or os.environ.get("taskdate")
```
Comment (gridcell) [inline task_base.py:46]: could do ...

`_taskdate = kwargs.get("taskdate") or os.environ.get("taskdate") or datetime.now(timezone.utc).date()`

File: task_base.py [hunk 2]
```diff
@@ -503,20 +504,22 @@ def wait(self):
...
+                    ee_task_state = s["state"]
+                    ee_task_id = s["id"]
+
+                    if ee_task_state in self.EEFINISHED:
+                        if ee_task_state == self.EEFAILED:
+                            self._failed_ee_tasks[ee_task_id] = s
+                        del self.ee_tasks[ee_task_id]
+                    else:
+                        self.ee_tasks[s["id"]] = s
+            except ConnectionResetError:
```
Comment (gridcell) [inline task_base.py:566]: Does ConnectionResetError need to be imported

File: task_base.py (hunk 1 above)
Comment (saanobhaai) [reply task_base.py:46]: Can't because the kwargs/env values are strings, not a date

File: task_base.py (hunk 2 above)
Comment (saanobhaai) [reply task_base.py:566]: Apparently it's a builtin, so I don't think so. I don't get an import error when running, either.

Comment (gridcell) [reply task_base.py:566]: So it is, haven't used that one

**Review by gridcell — APPROVED** [review]

**Pull request merged**